library/common/extensions/filters/http/platform_bridge/filter.cc

```diff
@@ -313,9 +321,111 @@ PlatformBridgeFilter::encodeTrailers(Http::ResponseTrailerMap& trailers) {
   if (status == Http::FilterTrailersStatus::StopIteration) {
     pending_response_trailers_ = &trailers;
   }
+  response_complete_ = true;
   return status;
 }
 
+void PlatformBridgeFilter::onResumeDecoding() {
+  Buffer::Instance* internal_buffer = nullptr;
+  if (decoder_callbacks_->decodingBuffer()) {
+    decoder_callbacks_->modifyDecodingBuffer([&internal_buffer](Buffer::Instance& mutable_buffer) {
+      internal_buffer = &mutable_buffer;
+    });
+  }
+
+  envoy_headers* pending_headers = nullptr;
+  envoy_data* pending_data = nullptr;
+  envoy_headers* pending_trailers = nullptr;
+  if (pending_request_headers_) {
+    *pending_headers = Http::Utility::toBridgeHeaders(*pending_request_headers_);
+  }
+  if (internal_buffer) {
+    *pending_data = Buffer::Utility::copyToBridgeData(*internal_buffer);
+  }
+  if (pending_request_trailers_) {
+    *pending_trailers = Http::Utility::toBridgeHeaders(*pending_request_trailers_);
+  }
+
+  envoy_filter_resume_status result =
+      platform_filter_.on_resume_request(pending_headers, pending_data, pending_trailers,
+                                         request_complete_, platform_filter_.instance_context);
+  if (result.status == kEnvoyFilterResumeStatusStopIteration) {
+    return;
+  }
+  if (pending_request_headers_) {
+    PlatformBridgeFilter::replaceHeaders(*pending_request_headers_, *result.pending_headers);
+    pending_request_headers_ = nullptr;
+    free(result.pending_headers);
+  }
```

Comment:
Should we assert in the `else` case that `result.pending_headers` is not passed?

Reply:
Sure. It will explode either way, but an `ASSERT` lets us give a nice reason for it. I'll update.